solidity/contracts/acreBTC.sol

```diff
@@ -256,47 +303,94 @@ contract acreBTC is ERC4626Fees, PausableOwnable {
+    /// @dev This function routes through the withdrawal queue for redemptions.
+    /// @param shares Amount of shares to redeem.
     /// @param receiver The address to which the assets will be transferred.
     /// @param owner The address of the owner of the shares.
-    function withdraw(
-        uint256 assets,
+    /// @return requestId The ID of the withdrawal request in the queue.
+    function requestRedeem(
+        uint256 shares,
         address receiver,
         address owner
-    ) public override returns (uint256) {
-        uint256 currentAssetsBalance = IERC20(asset()).balanceOf(address(this));
-        // If there is not enough assets in the vault to cover user withdrawals and
-        // withdrawal fees then pull the assets from the dispatcher.
-        uint256 assetsWithFees = assets + _feeOnRaw(assets, exitFeeBasisPoints);
-        if (assetsWithFees > currentAssetsBalance) {
-            dispatcher.withdraw(assetsWithFees - currentAssetsBalance);
+    ) external returns (uint256 requestId) {
+        if (withdrawalQueue == address(0)) {
+            revert WithdrawalQueueNotSet();
+        }
+
+        uint256 exitFee = _feeOnTotal(
+            convertToAssets(shares),
+            _exitFeeBasisPoints()
+        );
+
+        // Transfer shares to withdrawal queue using internal _transfer
+        if (msg.sender != owner) {
+            _spendAllowance(owner, msg.sender, shares);
         }
 
-        return super.withdraw(assets, receiver, owner);
+        _transfer(owner, withdrawalQueue, shares);
+
+        // Process redemption through queue
+        requestId = WithdrawalQueue(withdrawalQueue).requestRedeem(
+            shares,
+            receiver,
+            exitFee
+        );
+
+        emit RedemptionRequested(
+            requestId,
+            owner,
+            receiver,
+            msg.sender,
+            shares
+        );
     }
 
-    /// @notice Redeems shares for assets and transfers them to the receiver.
-    /// @dev Redeem unallocated assets first and and if not enough, then pull
-    ///      the assets from the dispatcher.
+    /// @notice Burns shares from the caller.
+    /// @param shares Amount of shares to burn.
+    function burn(uint256 shares) public {
+        if (msg.sender != address(withdrawalQueue)) {
+            revert OnlyWithdrawalQueue();
+        }
+
+        _burn(msg.sender, shares);
+    }
+
+    /// @notice Redeems shares for Bitcoin bridge withdrawal.
+    /// @dev This function routes through the withdrawal queue for bridge redemptions.
     /// @param shares Amount of shares to redeem.
-    /// @param receiver The address to which the assets will be transferred.
-    /// @param owner The address of the owner of the shares.
-    function redeem(
+    /// @param redeemerOutputScript Redeemer output script.
+    /// @return requestId The ID of the withdrawal request in the queue.
+    function requestRedeemAndBridge(
         uint256 shares,
```

Comment:
There is no guarantee the provided shares amount represents enough tBTC so that requesting to-BTC redemption on the bridge is possible.